File: src/main/java/com/github/cybellereaper/fusionBlock/nano/NanoOutEvent.java
package com.github.cybellereaper.fusionBlock.nano;

import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

public class NanoOutEvent extends Event implements Cancellable {
    private static final HandlerList HANDLERS = new HandlerList();
    private boolean cancelled;

    public NanoOutEvent() {
        super(true);
    }

    @Override
    public boolean isCancelled() {
        return cancelled;
    }

    @Override
    public void setCancelled(boolean cancelled) {
        this.cancelled = cancelled;
    }

    @Override
    public HandlerList getHandlers() {
        return HANDLERS;
    }

    public static HandlerList getHandlerList() {
        return HANDLERS;
    }
}

File: src/main/kotlin/com/github/cybellereaper/fusionBlock/nano/NanoOutEvent.kt
package com.github.cybellereaper.fusionBlock.nano

import org.bukkit.event.Cancellable
import org.bukkit.event.HandlerList

class NanoOutEvent : Cancellable {
    private var isCancelled: Boolean = false

    companion object {
        @JvmStatic
        val handlerList: HandlerList = HandlerList()
    }

    override fun isCancelled(): Boolean {
        return isCancelled
    }

    override fun setCancelled(p0: Boolean) {
        this.isCancelled = p0
    }
}